include/posix/pthread.h

```diff
@@ -193,7 +212,7 @@ static inline int pthread_mutex_timedlock(pthread_mutex_t *m,
 {
 	int ret = k_sem_take(m->sem, _ts_to_ms(to));
 
-	return ret == -EAGAIN ? -ETIMEDOUT : ret;
+	return ret == EAGAIN ? ETIMEDOUT : ret;
```

Comment:
k_sem_take() will return a negative error code. Error codes should be translated correctly here, you can't just return whatever it returns.

Reply:
In fact, this is the same for other APIs here (I can't comment on pthread_mutex_lock(), etc).  I did comment on this a few times already, and even provided code in comments to help convert the error codes.

Reply:
k_sem_take is going to return 
    1. 0 --> sucess
    2. -EAGAIN or -EBUSY 
So return value of "pthread_mutex_timedlock" i have modified  to 
return ret != 0 ? ETIMEDOUT : ret;